src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double FACTOR_SIZE = 1.5;
+    private static final int DEFAULT_SIZE = 10;
+    private T[] data = (T[]) new Object[DEFAULT_SIZE];
```

Comment:
you should initialize this inside the constructor
`private T[] data = (T[]) new Object[DEFAULT_SIZE];`